### ISSUE TITLE
New Training With Sliced Prediction PSPNet is Very Slow

### ISSUE THREAD

**Issue by isaykatsman** [ZijunDeng/pytorch-semantic-segmentation#11]
Hi, thanks for your implementation Zijun. Before, I was also obtaining ~60% mIoU for PSPNet segmentation of cityscapes, much lower than the result reported in the paper, and I think the sliced prediction will help improve this metric. However, epochs run very slowly now. Even with training batch size at 2 (as much as 12 GB on a K40 allows), an epoch takes nearly 12 hours! (whereas before it took maybe 30 minutes). Is this the same case with the training you run?

**Comment by ZijunDeng** [comment]
It takes about 4 hours (mush slower than before!!!) to train an epoch (batch size=2 on 1080Ti). The code on github is not latest and I will update it later.

**Comment by isaykatsman** [comment]
Thanks, I see. Also it seems that you added `sliced_forward` to repo, but you are not yet using it for evaluation? Same with `SlidingCropTest`?

**Comment by ZijunDeng** [comment]
At first I wanted to implement sliced prediction by just adding a decorator to the forward function of network. But during training my GPU cannot hold all the slices of one image, so I finally didn't use it. The `SlidingCropTest` on the github is not finished, I will push my latest code to github in several days.

**Comment by isaykatsman** [comment]
I see - makes sense. Thank you for your efforts! :)
